core/src/main/java/org/apache/carbondata/core/indexstore/blockletindex/BlockDataMap.java

```diff
@@ -117,9 +117,16 @@
     BlockletDataMapModel blockletDataMapInfo = (BlockletDataMapModel) dataMapModel;
     DataFileFooterConverter fileFooterConverter =
         new DataFileFooterConverter(dataMapModel.getConfiguration());
-    List<DataFileFooter> indexInfo = fileFooterConverter
-        .getIndexInfo(blockletDataMapInfo.getFilePath(), blockletDataMapInfo.getFileData(),
-            blockletDataMapInfo.getCarbonTable().isTransactionalTable());
+    List<DataFileFooter> indexInfo = null;
+    if (blockletDataMapInfo.getIndexInfos() == null || blockletDataMapInfo.getIndexInfos()
+        .isEmpty()) {
+      indexInfo = fileFooterConverter
+          .getIndexInfo(blockletDataMapInfo.getFilePath(), blockletDataMapInfo.getFileData(),
+              blockletDataMapInfo.getCarbonTable().isTransactionalTable());
+    } else {
+      // when index info is already read and converted to data file footer object
+      indexInfo = blockletDataMapInfo.getIndexInfos();
```

Comment:
Please reverify whether we are keeping BlockletDataMapModel  inside memory

Reply:
Not required as we are not keeping model in memory